examples/sandboxer.rs

```diff
@@ -0,0 +1,141 @@
+    paths: OsString,
+    access: AccessFs,
+) -> Result<Ruleset, anyhow::Error> {
+    let mut ruleset = ruleset;
+    let mut actual_access = access.clone();
+    let individual_paths: Vec<PathBuf> = paths
+        .into_vec()
+        .split(|b| *b == b':')
+        .map(|v| OsStr::from_bytes(v).to_owned().into())
+        .collect();
+
+    for path in individual_paths {
+        match open(&path, OFlag::O_PATH | OFlag::O_CLOEXEC, Mode::empty()) {
+            Err(e) => {
+                bail!("Failed to open \"{}\": {}\n", path.to_string_lossy(), e);
+            }
+            Ok(parent_fd) => match fstat(parent_fd) {
+                Ok(stat) => {
+                    if (stat.st_mode & SFlag::S_IFMT.bits()) != SFlag::S_IFDIR.bits() {
+                        actual_access.insert(ACCESS_FILE);
+                    }
+
+                    ruleset = match ruleset.add_path_beneath_rule(parent_fd, actual_access) {
+                        Ok(r) => r,
+                        Err(e) => {
+                            bail!(
+                                "Failed to update ruleset with \"{}\": {}\n",
+                                path.to_string_lossy(),
+                                e
+                            );
+                        }
+                    };
+                }
+                Err(e) => {
+                    bail!("Failed to stat \"{}\": {}\n", path.to_string_lossy(), e);
+                }
+            },
+        }
+    }
+
+    Ok(ruleset)
+}
+
+fn main() -> Result<(), anyhow::Error> {
+    let args: Vec<_> = env::args_os().collect();
+
+    let program_name = args.get(0).map(|s| s.to_string_lossy()).unwrap_or_default();
+
+    if args.len() < 2 {
+        eprintln!(
+            "usage: {}=\"...\" {}=\"...\" {} <cmd> [args]...\n",
+            ENV_FS_RO_NAME, ENV_FS_RW_NAME, program_name
+        );
+        eprintln!("Launch a command in a restricted environment.\n");
+        eprintln!("Environment variables containing paths, each separated by a colon:");
+        eprintln!(
+            "* {}: list of paths allowed to be used in a read-only way.",
+            ENV_FS_RO_NAME
+        );
+        eprintln!(
+            "* {}: list of paths allowed to be used in a read-write way.",
+            ENV_FS_RW_NAME
+        );
+        eprintln!(
+            "\nexample:\n\
+                {}=\"/bin:/lib:/usr:/proc:/etc:/dev/urandom\" \
+                {}=\"/dev/null:/dev/full:/dev/zero:/dev/pts:/tmp\" \
+                {} bash -i",
+            ENV_FS_RO_NAME, ENV_FS_RW_NAME, program_name
+        );
+        return Ok(());
+    }
+
+    let fs_ro = env::var_os(ENV_FS_RO_NAME)
+        .expect(&format!("Missing environment variable {}", ENV_FS_RO_NAME));
+    let fs_rw = env::var_os(ENV_FS_RW_NAME)
+        .expect(&format!("Missing environment variable {}", ENV_FS_RW_NAME));
+
+    let cmd_name = args
+        .get(1)
+        .map(|s| s.to_string_lossy())
+        .unwrap_or(Cow::Borrowed("bash -i"));
```

Comment:
This is unfortunately not reached because the `args.len()` check error out before.

Reply:
I am not sure I understand, `cmd_name` here is the command requested by the user e.g. `bash`. It corresponds to [sandboxer.c:226](https://github.com/landlock-lsm/linux/commit/e3f0859b73352f2beae6eae1ecd356355137f644#diff-d57a3a6675aea6e1eefd15ad854d8f416fa334eae25cd7b379fc5cf3d5f9506eR226).

Reply:
`cmd_name` is correct, I was talking about the default value (cf. other comment).